divide_and_conquer/binary_exponentiation.cpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * @file
+ * @brief Implementation of Binary Exponentiation
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂